src/atomate2/vasp/flows/defect.py

```diff
@@ -22,28 +30,127 @@
 
 logger = logging.getLogger(__name__)
 
+################################################################################
+# Default settings                                                            ##
+################################################################################
 
 DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
 DEFECT_INCAR_SETTINGS = {
     "ISMEAR": 0,
-    "LWAVE": True,
     "SIGMA": 0.05,
     "KSPACING": None,
     "ENCUT": 500,
 }
-DEFECT_KPOINT_SETTINGS = {"reciprocal_density": 64}
+DEFECT_KPOINT_SETTINGS = Kpoints(
+    comment="special k-point",
+    num_kpts=1,
+    style=Kpoints_supported_modes.Reciprocal,
+    kpts=((0.25, 0.25, 0.25),),
+    kpts_shift=(0, 0, 0),
+    kpts_weights=[1],
+)
 
-DEFECT_RELAX_GENERATOR = AtomicRelaxSetGenerator(
+DEFECT_RELAX_GENERATOR: AtomicRelaxSetGenerator = AtomicRelaxSetGenerator(
     use_structure_charge=True,
     user_incar_settings=DEFECT_INCAR_SETTINGS,
     user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
 )
-DEFECT_STATIC_GENERATOR = StaticSetGenerator(
+
+DEFECT_STATIC_GENERATOR: StaticSetGenerator = StaticSetGenerator(
     user_incar_settings=DEFECT_INCAR_SETTINGS,
     user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
 )
 
 
+################################################################################
+# Formation Energy                                                            ##
+################################################################################
+
+
+@dataclass
+class FormationEnergyMaker(Maker):
+    """Class to generate VASP input sets for the calculation of the formation energy
+    diagram.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow created by this maker.
+    relax_maker: .BaseVaspMaker or None
+        A maker to perform a atomic-position-only relaxation on the defect charge
+        states.
+        If None, the defaults will be used.
+    """
+
+    name: str = "formation energy"
+
+    relax_maker: BaseVaspMaker = field(
+        default_factory=lambda: RelaxMaker(
+            input_set_generator=DEFECT_RELAX_GENERATOR,
+            task_document_kwargs={"store_volumetric_data": ["locpot"]},
+        )
+    )
+
+    def __post_init__(self):
+        """Post-initialization."""
+        self.relax_maker = self.relax_maker.update_kwargs(
+            update={
```

Comment:
One issue with this, is that any user settings will be overridden. I know if a user changes LVHAR this could break downstream things, however, if a user has specifically overridden something then we should defer to them as they might have a particular reason for doing so.

Instead, you could add these settings to the `DEFECT_RELAX_GENERATOR`, and that will also make this class a bit cleaner.

Reply:
OK, I would still like to add some kind of checking and warning mechanism here, that might go in the different PR.

Reply:
I made the following PR to jobflow that will allow me to do some paranoia checks on the supplied Maker.
https://github.com/materialsproject/jobflow/pull/177
It is getting a bit complicated here because for long-running HSE calculations I found that doing a PBE relaxation helps save a lot of time and tends a avoid charge sloshing.  So I am now feeding the maker a DoubleRelaxMaker with PBE followed by HSE.

Reply:
So I commented this out and will be added back after the PR is made.